src/dnscap.c

```diff
@@ -1099,6 +1130,11 @@ parse_args(int argc, char *argv[]) {
 		usage("the -L and -l options are mutually exclusive");
 	if (background && (dumptrace || preso))
 		usage("the -b option is incompatible with -d and -g");
+        if (sample && chooseSidesResponse)
+        {
+                if (!((dir_wanted & DIR_INITIATE) != 0) && ((dir_wanted & DIR_RESPONSE) != 0))
```

Comment:
It should be incompatible with -s r. I fixed this flaw.
I'm actually sorry, I thought I had tested this..